mailbox/store/src/main/java/org/apache/james/mailbox/store/mail/model/impl/MessageParser.java

```diff
@@ -178,19 +196,8 @@ public String apply(String input) {
     }
 
     private Optional<Cid> cid(Optional<ContentIdField> contentIdField) {
-        return contentIdField.transform(new Function<ContentIdField, Optional<Cid>>() {
-            @Override
-            public Optional<Cid> apply(ContentIdField field) {
-                return Optional.fromNullable(field.getId())
-                        .transform(new Function<String, Cid>() {
-
-                            @Override
-                            public Cid apply(String cid) {
-                                return Cid.from(cid);
-                            }
-                        });
-            }
-        }).or(Optional.<Cid> absent());
+        return contentIdField.transform(CONTENT_ID_FIELD_TO_OPTIONAL_CID_FUNCTION)
+                .or(Optional.<Cid> absent());
```

Comment:
It's exactly what I thought but no what she is doing here is a flatMap but there is no flatMap in Guavate. Before the or she has a Optional<Optional<ContIdField>> she want a Optional<ContentIdField> if the first level is present then you just want it's inner value otherwise you want absent anyway therefore is this .or